presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -498,6 +498,13 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                 }
```

Comment:
this error is not specific to dynamic filters. The comment before PPD in `PlanOptimizers` says:
```
        // Run predicate push down one more time in case we can leverage new information from layouts' effective predicate
```
It might happen that these new predicates could also cause outer join to convert to inner join in similar way. This is general problem with PPD.

Reply:
yes, the problem in general is not DF specific. But is more easily reproducible with DF, because DF processing in PPD is delayed (https://github.com/prestosql/presto/commit/a5eb69529de6915e02d8a1ccd5cbf0a56a0b277b)

> This is general problem with PPD and existed before

yes